x-pack/legacy/plugins/alerting/server/builtin_alert_types/index_threshold/index.ts

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { Logger } from '../../../../../../../src/core/server';
+
+import { AlertType, AlertExecutorOptions } from '../../types';
+import { Params, ParamsSchema } from './params';
+import { BaseActionContext, addMessages } from './action_context';
+
+import { runQuery } from '../../../../../../plugins/alerting_index_threshold/server';
+
+export const ID = '.index-threshold';
+
+interface GetAlertTypeParams {
+  logger: Logger;
+}
+
+const ActionGroup = 'threshold met';
+const CompareFns = getCompareFns();
+
+export function getAlertType(getAlertTypeParams: GetAlertTypeParams): AlertType {
+  const { logger } = getAlertTypeParams;
+
+  const alertTypeName = i18n.translate('xpack.alerting.builtin.indexThresholdTitle', {
+    defaultMessage: 'Index Threshold',
+  });
+
+  return {
+    id: ID,
+    name: alertTypeName,
+    actionGroups: [ActionGroup],
+    validate: {
+      params: ParamsSchema,
+    },
+    executor,
+  };
+
+  async function executor(options: AlertExecutorOptions) {
+    const { alertId, name, services } = options;
+    const params: Params = options.params as Params;
+
+    const compareFn = CompareFns[params.comparator];
+    if (compareFn == null) {
+      throw new Error(`invalid comparator specified: "${params.comparator}"`);
+    }
+
+    const callCluster = services.callCluster;
+    const date = new Date().toISOString();
+    const queryParams = {
+      index: params.index,
+      timeField: params.timeField,
+      aggType: params.aggType,
+      aggField: params.aggField,
+      groupField: params.groupField,
+      groupLimit: params.groupLimit,
+      date,
+      interval: params.window,
```

Comment:
`interval` is actually ignored in the case of `intervalsBefore +intervalsAfter === 1`.  Having to send _something_ is awkward in this case, seems like it should be optional ... would be a validation error to have total intervals != 1 and `interval` not set ...